qiskit/providers/backend_compat.py

```diff
@@ -187,18 +214,17 @@ def convert_to_target(
             if qubit_idx in faulty_qubits:
                 continue
             qubit_prop = properties.qubit_property(qubit_idx)
-            in_prop = {
-                "duration": qubit_prop["readout_length"][0]
-                if "readout_length" in qubit_prop
-                else None,
-                "error": qubit_prop["readout_error"][0] if "readout_error" in qubit_prop else None,
-            }
-            prop_name_map["measure"][(qubit_idx,)] = InstructionProperties(**in_prop)
+            prop_name_map["measure"][(qubit_idx,)] = InstructionProperties(
+                error=_get_value(qubit_prop, "readout_error"),
+                duration=_get_value(qubit_prop, "readout_length"),
+            )
 
-    if add_delay and "delay" not in prop_name_map:
-        prop_name_map["delay"] = {
-            (q,): None for q in range(configuration.num_qubits) if q not in faulty_qubits
-        }
+    for op in required:
+        # Map required ops to each operational qubit
+        if prop_name_map[op] is None:
+            prop_name_map[op] = {
+                (q,): None for q in range(configuration.num_qubits) if q not in faulty_qubits
```

Comment:
Shouldn't we only exclude the faulty qubits here if `filter_faulty` is `True`?

Reply:
Done in [bcb4144](https://github.com/Qiskit/qiskit/pull/11404/commits/bcb4144567c2ea92714cc01e5241b2b7e9c3f557) (also added a test).